core/src/main/scala/io/finch/syntax/scala/ScalaToTwitterFuture.scala

```diff
@@ -1,21 +1,30 @@
 package io.finch.syntax.scala
 
-import scala.concurrent.{ExecutionContext, Future => ScalaFuture}
+import scala.concurrent.{Future => ScalaFuture}
 import scala.util.{Failure, Success}
 
 import com.twitter.util.{Future, Promise}
 import io.finch.syntax.ToTwitterFuture
 
 trait ScalaToTwitterFuture {
-  implicit def scalaToTwitterFuture(implicit ec: ExecutionContext): ToTwitterFuture[ScalaFuture] =
+  implicit lazy val scalaToTwitterFuture: ToTwitterFuture[ScalaFuture] =
     new ToTwitterFuture[ScalaFuture] {
       def apply[A](f: ScalaFuture[A]): Future[A] = {
-        val p = Promise[A]
-        f.onComplete {
-          case Success(a) => p.setValue(a)
-          case Failure(t) => p.setException(t)
+        if (f.isCompleted) {
```

Comment:
I'd prefer we kill this optimization as it's premature.

Reply:
Ok. It's unlikely to make a big difference under real-world circumstances anyway (i.e. futures that do more than return a static value).